thrust/thrust/system/cuda/detail/unique.h

```diff
@@ -634,8 +634,9 @@ unique_count(execution_policy<Derived>& policy, ForwardIt first, ForwardIt last,
     return 0;
   }
   auto size = thrust::distance(first, last);
-  auto it   = thrust::make_zip_iterator(thrust::make_tuple(first, thrust::next(first)));
-  return 1 + thrust::count_if(policy, it, thrust::next(it, size - 1), zip_adj_not_predicate<BinaryPred>{binary_pred});
+  auto it   = thrust::make_zip_iterator(thrust::make_tuple(first, ::cuda::std::next(first)));
```

Comment:
@miscco are we also moving `make_zip_iterator` and `make_tuple` to libcu++?

Reply:
We cannot currently move `zip_iterator` because that has no equivalent. 

I do want to get rid of `thrust::tuple` internally, but there be dragons